btc/usd reporter example

oraclown opened the pull request:
Runs an example datafeed for BTC/USD. Pulls latest value to submit from local db. Submitting on-chain still needs to be implemented.

Usage: in src/, run `python -m telliot report`

Visit `http://127.0.0.1:8000/data/` to view off-chain data stored locally.
Visit `http://127.0.0.1:8000/docs/` to test the database API in the browser.

clamdad commented:
I am getting an error running on windows - something with multiprocessing module when the database starts up.

clamdad commented:
It seems like the database object should live somewhere in the database package (e.g. a new class `DataSourceDb`), rather than an attribute of the `Reporter`.  This way, the database could be started separately, even as part of a separate python interpreter instance.

clamdad commented:
We need to think about the runtime environment and if the top-level Scheduler task should run as a thread, an [asyncio.task](https://docs.python.org/3/library/asyncio-task.html#task-object) (or both - only one asyncio event loop per thread).  I'm not sure, but in either case, it should be possible to initiate and cleanly shutdown the task from a higher layer of software, rather than relying on a keyboard interrupt.   So in either case, it should have a safe termination event defined.  In the case of asyncio, it would respond to a cancel event by handling a `CancelledError` exception.  For a thread, we can create our own threading.Event. 

Same idea for the top level database task.

clamdad commented on src/telliot/reporter.py:
```diff
@@ -0,0 +1,64 @@
+        try:
+            # TODO: Run asyncio loop forever.
+            pass
+        except (KeyboardInterrupt, SystemExit):
```
Should be able to terminate from software using an event.

oraclown commented:
Removed all the database stuff from the rinkeby-btc-usd example, per @clamdad's earlier comments & our discussion with @tallywiesenberg. Also, to keep the example as lean as possible.

Moved some more of the example code into the `plugins` directory.

The cli now runs this btc price reporter if the `report` flag is used, you just need to rename `.env.example` to `.env` and add your private key and node endpoint url. Then in /src run `python -m telliot report`.

I'm not attached to this way of organizing plugins/examples, so let me know y'all's thoughts.

oraclown commented:
changed name of plugins folder to `reporter_plugins` per @tallywiesenberg's point

clamdad commented:
* RinkebySubmitter: I don't think we need a subclass here.  We could just give the Submitter endopint attribute, and just pass in a different endpoint when we create the instance.   The endpoint could be something like that in #28.
* The reporter_plugins structure is confusing to me.  I'm not sure what we would be asking the user to do.  Still seems early/premature to force a specific plugin architecture.
* Not a fan of using dotenv or OS environment variables to pass configuration options.  Especially for private keys.   There are  better options, like the built in configparser module or yaml or json. 
* Config files should be put in a standard location rather than in the repository tree.  I suggest something like `pathlib.Path.home() / '.pytelliot'`
